libzfs.pyx

```diff
@@ -1076,6 +1091,34 @@ cdef class ZFS(object):
         datasets = datasets or [p.name for p in self.pools]
         return ZFS._snapshots_serialized_impl(self.handle, datasets, props, holds, mounted, recursive, min_txg, max_txg)
 
+    @staticmethod
+    cdef int _userspace_cb(void *data, const char *domain, uint32_t rid, uint64_t space) nogil:
```

Comment:
Is there a reason to have function definition as `nogil` but inside it grabs the `gil` always?

Reply:
I think you can remove the `nogil` and then remove `with gil` statement so this is done while gil is held.

Reply:
I saw it done this way somewhere else, and I like that it's explicitly grabbing the gil inside this function as it is a callback invoked from a C context.